packages/cli/src/linter/integration-linter.test.ts

```diff
@@ -0,0 +1,235 @@
+  readme: TRUTHY_STRING,
+  icon: TRUTHY_STRING,
+  actions: {
+    [ACTION_NAME]: {
+      title: TRUTHY_STRING,
+      description: TRUTHY_STRING,
+      input: {
+        schema: z
+          .object({
+            [PARAM_NAME]: z.string().title(TRUTHY_STRING).describe(TRUTHY_STRING),
+          })
+          .title(TRUTHY_STRING)
+          .describe(TRUTHY_STRING),
+      },
+      output: {
+        schema: z
+          .object({
+            [PARAM_NAME]: z.string().title(TRUTHY_STRING).describe(TRUTHY_STRING),
+          })
+          .title(TRUTHY_STRING)
+          .describe(TRUTHY_STRING),
+      },
+    },
+  },
+  events: {
+    [EVENT_NAME]: {
+      title: TRUTHY_STRING,
+      description: TRUTHY_STRING,
+      schema: z
+        .object({
+          [PARAM_NAME]: z.string().title(TRUTHY_STRING).describe(TRUTHY_STRING),
+        })
+        .title(TRUTHY_STRING)
+        .describe(TRUTHY_STRING),
+    },
+  },
+  configuration: {
+    schema: z
+      .object({
+        [PARAM_NAME]: z.string().title(TRUTHY_STRING).describe(TRUTHY_STRING),
+      })
+      .title(TRUTHY_STRING)
+      .describe(TRUTHY_STRING),
+  },
+  configurations: {
+    [CONFIG_NAME]: {
+      title: TRUTHY_STRING,
+      description: TRUTHY_STRING,
+      schema: z
+        .object({
+          [PARAM_NAME]: z.string().title(TRUTHY_STRING).describe(TRUTHY_STRING),
+        })
+        .title(TRUTHY_STRING)
+        .describe(TRUTHY_STRING),
+    },
+  },
+  user: {
+    tags: {
+      [TAG_NAME]: {
+        title: TRUTHY_STRING,
+        description: TRUTHY_STRING,
+      },
+    },
+  },
+  channels: {
+    [CHANNEL_NAME]: {
+      title: TRUTHY_STRING,
+      description: TRUTHY_STRING,
+      messages: {
+        [MESSAGE_TYPE]: {
+          schema: z
+            .object({
+              [PARAM_NAME]: z.string().title(TRUTHY_STRING).describe(TRUTHY_STRING),
+            })
+            .title(TRUTHY_STRING)
+            .describe(TRUTHY_STRING),
+        },
+      },
+      message: {
+        tags: {
+          [TAG_NAME]: {
+            title: TRUTHY_STRING,
+            description: TRUTHY_STRING,
+          },
+        },
+      },
+      conversation: {
+        tags: {
+          [TAG_NAME]: {
+            title: TRUTHY_STRING,
+            description: TRUTHY_STRING,
+          },
+        },
+      },
+    },
+  },
+  states: {
+    [STATE_NAME]: {
+      type: 'integration',
+      schema: z
+        .object({
+          [PARAM_NAME]: z.string().title(TRUTHY_STRING).describe(TRUTHY_STRING),
+        })
+        .title(TRUTHY_STRING)
+        .describe(TRUTHY_STRING),
+    },
+  },
+} as const satisfies IntegrationDefinitionProps
+
+const mockLogger = {
+  log: vi.fn((message) => void message),
+  warn: vi.fn((message) => void message),
+  error: vi.fn((message) => void message),
+  debug: vi.fn((message) => void message),
+}
+
+const lintDefinition = async (definition: ConstructorParameters<typeof IntegrationDefinition>[0]) => {
```

Comment:
**nitpick:** there's already a type for that called `IntegrationDefinitionProps`